app/javascript/src/controller_pages.js

```diff
@@ -215,23 +216,28 @@ function addQuestionMenuListeners(view) {
 
   // QuestionMenuSelectionRemove
   view.$document.on("QuestionMenuSelectionRemove", function(event, question) {
-    var html = $(templateContent).filter("[data-node=remove]").text();
-    view.dialogConfirmationDelete.open({
-      heading: html.replace(/#{label}/, question.$heading.text()),
-      ok: view.text.dialogs.button_delete_component
-      }, function() {
+    var questionUuid = question.data._uuid;
+    var apiUrl = question.menu.selectedItem.data('apiPath').replace('question_uuid', questionUuid);
+    
+    new DialogApiRequest(apiUrl, {
+      activator: question.menu.selectedItem,
+      closeOnClickSelector: ".govuk-button",
+
+      build: function(dialog) {
+        // Find and correct (make work!) any method:delete links
+        dialog.$node.find("[data-method=delete]").on("click", function(e) {
+          e.preventDefault();
       // Workaround solution that doesn't require extra backend work
       // 1. First remove component from view
```

Comment:
Like the attention to detail on indenting code but, is there a reason you didn't indent the comment as well?
<img width="618" alt="Screenshot 2022-01-26 at 12 15 16" src="https://user-images.githubusercontent.com/76942244/151161335-f5d177fa-e8fc-4f90-a5e3-b6fefc0262de.png">